### PR TITLE
add calCosDst in cal_groundtruth.cpp

### PR DESCRIPTION
**Pull request by ysong4**
Keep calL2Dst() as default.

### CODE REVIEW

File: apps/cal_groundtruth.cpp [hunk 1]
```diff
@@ -83,8 +84,27 @@ class Query {
             return sqrt(l2Dst);
         }
 
+        float calCosDst(const vector<float>& item) {
+            float cosDst = 0;
```
Comment (lijinf2) [inline apps/cal_groundtruth.cpp]: move the declaration of cosDst to where it is firstly used, i.e. line 99

File: apps/cal_groundtruth.cpp [hunk 2]
```diff
@@ -83,8 +84,27 @@ class Query {
...
+            float iNorm = 0;
+            for (int i = 0; i < this->content.size(); ++i) {
+                qNorm += this->content[i] * this->content[i];
+                iNorm += item[i] * item[i];
+            }
+            qNorm = sqrt(qNorm);
+            iNorm = sqrt(iNorm);
+            float combined = qNorm * iNorm;
+            for (int i = 0; i < this->content.size(); ++i) {
+                cosDst += this->content[i] * item[i] / combined;
```
Comment (lijinf2) [inline apps/cal_groundtruth.cpp]: combined is divided by content.size() times, which is unnecessary. Please change this line of code to 

for (int i = 0; i < this->content.size(); ++i) {
    cosDst += this->content[i] * item[i];
}

cosDst /= (qNorm * iNorm);

File: apps/cal_groundtruth.cpp [hunk 3]
```diff
@@ -83,8 +84,27 @@ class Query {
...
+            assert(this->content.size() == item.size());
+            float qNorm = 0;
+            float iNorm = 0;
+            for (int i = 0; i < this->content.size(); ++i) {
+                qNorm += this->content[i] * this->content[i];
+                iNorm += item[i] * item[i];
+            }
+            qNorm = sqrt(qNorm);
+            iNorm = sqrt(iNorm);
+            float combined = qNorm * iNorm;
```
Comment (lijinf2) [inline apps/cal_groundtruth.cpp]: Remove line 98, variable combined is not necessary.

File: apps/cal_groundtruth.cpp (hunk 1 above)
Comment (ysong4) [reply apps/cal_groundtruth.cpp]: Did you mean move calCosDst declaration to:

void updateQueries(vector<Query*> queries, const vector<vector<float>>* itemsPtr, int itemStartIdx) {
    for (auto& query: queries) {
        for (int i = 0; i < itemsPtr->size(); ++i) {
            query->evaluate((*itemsPtr)[i], itemStartIdx + i);
            {ADD CODE HERE}
        }
    }
}

? Not very clear what is line 99. As in both old and new version, line 99 is inside a function...

**Pull request merged**